chapters/annotations.tex

```diff
@@ -227,35 +248,49 @@ \subsubsection{Variable Replacements}\label{variable-replacements}
+
+Links take the form \lstinline!%[$\mathit{text}$]($\mathit{link}$)!, where the \lstinline![$\mathit{text}$]! part is optional.  Text markup escape sequences don't apply inside the $\mathit{link}$, which can be in either of the following forms:
 \begin{itemize}
-\item A URI, such as
-\lstinline!https://github.com/modelica/ModelicaSpecification! or
-\lstinline!modelica://Modelica.Blocks!.
-\item A \lstinline!variable:<id>!, where \lstinline!<id>! is a component
-reference such as \lstinline!inertia1.w!.
-\item A \lstinline!plot:<id>!, where \lstinline!<id>! is the identifier of a
-\lstinline!Plot! in the current \lstinline!Figure!.
+% Use very short Modelica URI combined with carefully chosen wording below to deal with difficult line breaking.
+\item A URI.  Well established schemes such as \lstinline[language={[nocomment]modelica}]!https://github.com/modelica/ModelicaSpecification! or \lstinline[language={[nocomment]modelica}]!modelica://Modelica!, as well as lesser known schemes may be used.  (A tool that has no special recognition of a scheme can try sending the URI to the operating system for interpretation.)
+\item A \lstinline!variable:$\mathit{id}$!, where $\mathit{id}$ is a component reference in the form of \lstinline[language=grammar]!component-reference! in the grammar, such as \lstinline!inertia1.w!.
+\item A \lstinline!plot:$\mathit{id}$!, where $\mathit{id}$ is the identifier of a \lstinline!Plot! in the current \lstinline!Figure!.
 \end{itemize}
 
-When \lstinline![<text>]! is omitted, a Modelica tool is free to derive a default based on
-the \lstinline!<link>!.
+When \lstinline![$\mathit{text}$]! is omitted, a Modelica tool is free to derive a default based on the $\mathit{link}$.
+
+\begin{nonnormative}
+Note that for the character `]' to appear in $\mathit{text}$, it needs to be encoded as the escape sequence \lstinline!%]!, or it would be interpreted as the terminating delimiter of the \lstinline![$\mathit{text}$]!.
+
+Similarly, the closing parenthesis `)' must be handled with care in $\mathit{link}$ in order to not be interpreted as the terminating delimiter of the \lstinline!($\mathit{link}$)!.
+\begin{itemize}
+\item For a URI, a closing parenthesis must be URL encoded in order to not be interpreted as the end of the \lstinline!($\mathit{link}$)!.  For example, the URL in \lstinline[language={[nocomment]modelica}]!%(http://bad.org/(tryme))! is just \filename{http://bad.org/(tryme}, and the entire link is followed by a stray closing parenthesis.  To make it work, one has to use URL encoding: \lstinline[language={[nocomment]modelica}]!%(http://bad.org/%28tryme%29)! (using URL encoding of the opening parenthesis just for symmetry, and note that the \lstinline!%! of the percent-encoded sequences are not subject to text markup escape sequences).
```

Comment:
Using something that possibly could be a valid url as example is error-prone. Shouldn't we follow RFC 2606 and replace bad.org by example.org?

Reply:
Sure.  Fixed.  Resolving.